txn/txn_test.go

```diff
@@ -1065,3 +1070,34 @@ func (s *S) TestTxnQueuePreparing(c *C) {
 	}
 	c.Check(len(qdoc.Queue), Equals, expectedCount)
 }
+
+func (s *S) TestTxnQueueAddAndRemove(c *C) {
+	opts := txn.DefaultRunnerOptions()
+	opts.MaxTxnQueueLength = 10
+	s.runner.SetOptions(opts)
+	opInsert := []txn.Op{{
+		C:      "accounts",
+		Id:     0,
+		Insert: M{"balance": 0},
+	}}
+	opRemove := []txn.Op{{
+		C:      "accounts",
+		Id:     0,
+		Remove: true,
+	}}
+	err := s.runner.Run(opInsert, "", nil)
+	c.Assert(err, IsNil)
+	for n := 0; n < 10; n++ {
+		err = s.runner.Run(opRemove, "", nil)
+		c.Assert(err, IsNil)
+		err = s.runner.Run(opInsert, "", nil)
+		c.Assert(err, IsNil)
+	}
+	var qdoc txnQueue
+	err = s.accounts.FindId(0).One(&qdoc)
+	c.Assert(err, IsNil)
+	err = s.accounts.FindId(0).One(&qdoc)
```

Comment:
Is there are reason to find the doc twice? Or is this an error?

Reply:
just a copy paste error in the test, fixed.